Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/incomplete-test-creation-dialog.component.ts

```diff
@@ -0,0 +1,31 @@
+﻿import { Component, Inject, OnInit } from '@angular/core';
+import { MdDialog, MdDialogRef, MD_DIALOG_DATA } from '@angular/material';
+import { ActivatedRoute } from '@angular/router';
+import { TestService } from '../tests.service';
+import { Test } from '../tests.model';
+
+@Component({
+    moduleId: module.id,
+    selector: 'incomplete-test-creation',
+    templateUrl: 'incomplete-test-creation-dialog.html'
+})
+
+export class IncompleteTestCreationDialogComponent implements OnInit {
+    isQuestionMissing: boolean;
+    testId: number;
+    test: Test;
+
+    constructor(public testService: TestService, public dialogRef: MdDialogRef<IncompleteTestCreationDialogComponent>, @Inject(MD_DIALOG_DATA) public data: any, public route: ActivatedRoute) {
+        this.test = new Test();           
+    }
+
+    ngOnInit() {
+        this.test = this.data;
```

Comment:
again the same mistakes repeated. why the redundant assignment is required.? can't you use direct data object?